### ISSUE TITLE
Allow the SCM annotations to refer to an arbitrary commit instead of the current HEAD

### ISSUE THREAD

**Issue by patrys** [Microsoft/vscode#36893]
Currently numerous places (explorer, gutters, etc.) will display changes since the last commit. I believe it would be useful to allow to show the same annotations but related to a different branch or tag. I am currently working on a large pull request and find myself constantly running `git diff master` in a console.

**Comment by lolmaus** [comment]
Hi! I've requested the same feature in #61610, where I was pointed at this one. So happy to know that I'm not alone!

Providing my feature request below.

***

VS Code highlights changed lines against head branch, just like any other IDE does. This means that you can see only uncommitted edits.

I am typically working on a feature for several days. At the end of every day I need to commit my work in order to report my progress and back it up.

After I commit, I no longer see edits that I'm introducing with the given feature.

**I want changed lines to stay highlighted even after I commit**. This could be achieved by letting users select a branch to compare against (head by default).

As a workaround, I'm using [GitLens](https://marketplace.visualstudio.com/items?itemName=eamodio.gitlens) feature `Compare Ancestry with Working Tree` against the dev branch. This works, but it is a very elaborate action, and it shows results in a separate window. I want changes to be highlighted inline as I work on files.

Thank you for your attention. :bow:

**Comment by rynjwssl** [comment]
I'm just hopping in here to say I agree that this would be a killer feature. GitLens does get the job done, but if it were to be highlighted inline that would be a big improvement!

**Comment by SethRzeszutek** [comment]
Still hoping for this feature, it would be extremely useful in my workplace.

**Comment by mpbento** [comment]
Yes, I also believe this would be really useful. This is one of Atom fetaures that I am missing in vscode

**Comment by smelc** [comment]
As an alternative I open the commit I wanna seein [GitLens](https://marketplace.visualstudio.com/items?itemName=eamodio.gitlens) and highlight its changes in the editor (see the green vertical bar in the margin of the opened file):

![gitlens-hl](https://user-images.githubusercontent.com/634720/114881795-9e213900-9e03-11eb-8e17-1fa0a37053e3.png)

It's not super convenient though.

**Comment by eamodio** [comment]
@smelc I totally agree that it is not ideal, but at least from the GitLens point of view -- is there something else that can be done to improve that flow? If, so could you please open an issue on the GitLens repo for that. Thanks!

**Comment by smelc** [comment]
@eamodio> I don't see an obvious better UI/UX GitLens-wise for this :-| Thanks for your work on GitLens btw, great tool !:pray:

**Comment by morningmoni** [comment]
Is there any update on this feature?

**Comment by VVVi** [comment]
This would be a handy feature indeed.

**Comment by eamodio** [comment]
FWIW, once https://github.com/microsoft/vscode/issues/169012 lands, I plan to make the experience with GitLens around this better.

**Comment by IbrahimTanyalcin** [comment]
Are there any advancements in this feature request yet? This would be so handy and make life easier

**Comment by mstepin** [comment]
+1 on this feature. would be very handy. 🙏  please implement.

**Comment by bluegate010** [comment]
+1

**Comment by ferasterkawi** [comment]
+1 on this feature, sometimes I avoid do commits to keep some highlights #^_^#

**Comment by erbelion** [comment]
+1 i need this so much

**Comment by brandon-hills** [comment]
+1 for sure, would really like this improvement

**Comment by DanielHunte** [comment]
Please add this feature. VSCode kinda sucks without it.

**Comment by derekchang4** [comment]
+1 This would make keeping track of changes so much easier

**Comment by chrispab** [comment]
+1 would help

**Comment by Fmajor** [comment]
any updates for this feature?

**Comment by isalew** [comment]
+1 - One 💧 raises the 🌊... Hoping this eventually pulls through!

**Comment by sylvercode** [comment]
I would gladly in favor of the feature and drop my plug-in [gitbranchquickdiff](https://github.com/sylvercode/gitbranchquickdiff/tree/tree-view).
